src/committee/orchestrator.rs

```diff
@@ -36,12 +40,235 @@ use super::node::{Round2Request, Round2Response};
 // Orchestration logic
 //
 
+type RpcOrchestratorContext = Arc<(Orchestrator, Arc<RwLock<MemberStatusState>>)>;
+
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct CommitteeConfig {
     pub threshold: usize,
+    // TODO: We could use a Vec instead of a HashMap for the members, since it would be more efficient.
+    // We do not currently need hashmap functionality, but we might later, so left unchanged.
     pub members: HashMap<frost_secp256k1_tr::Identifier, Member>,
 }
 
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq)]
+pub enum MemberStatus {
+    Online,
+    /// Disconnected members contain a tuple with the next connect retry time and the last retry number
+    Disconnected((u64, u8)),
+    /// Will no longer retry
+    Offline,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct StatusResponse {
+    pub online_members: Vec<frost_secp256k1_tr::Identifier>,
+    pub offline_members: Vec<frost_secp256k1_tr::Identifier>,
+}
+
+// This will be the second part in the RpcModule context wrapped in a RwLock.
+// I think this is better than including it in the actual CommitteeConfig since handlers will need
+// to wait for a read lock every time an rpc handler needs to access the config.
+// This way, handlers will only wait for read locks when they need the status of the members.
+pub struct MemberStatusState {
+    pub key_to_addr: HashMap<frost_secp256k1_tr::Identifier, String>,
```

Comment:
Yup. Probably best if we moved the status logic to the Orchestrator struct so it would always have access to that map, but figured this would be better for separating this auxiliary functionality. The map is not going to grow large anyway.